### PR TITLE
Add pre-trained models from HuggingFace and add accuracy testing

### LINKED ISSUE #72: ResNet producing non deterministic and wrong predictions
Predictions from the ResNet pre-trained models seems to have an issue. The predicted probabilities are changing at each call and essentially look like random. VGG19 does work fine however.

```julia
using Flux
using Metalhead

img = Metalhead.load("data/cats/cats_00001.jpg");

julia> classify(VGG19(), img)
"tiger cat"

julia> classify(VGG19(), img)
"tiger cat"

julia> classify(ResNet(), img)
"reel"

julia> classify(ResNet(), img)
"abacus"

julia> classify(ResNet(), img)
"spotlight, spot"

julia> classify(ResNet(), img)
"fountain"
``` 

I first though it might be an issue with a different preprocessing required for ResNet, thought from the randomness in the output of the model, I'd guess some layers didn't get properly defined, possibly the BatchNorm?

If using `testmode!`, resnet will then produces the same predictions after each run. However, a different set of weights seem to be initialized right at each call to ``

```julia
resnet = ResNet().layers
testmode!(resnet)
cat_pred_resnet = resnet(preprocess(RGB.(img)))
julia> findmax(cat_pred_resnet)
(1.0f0, CartesianIndex(900, 1))

cat_pred_resnet = resnet(preprocess(RGB.(img)))
julia> findmax(cat_pred_resnet)
(1.0f0, CartesianIndex(900, 1))

resnet = ResNet().layers
testmode!(resnet)
cat_pred_resnet = resnet(preprocess(RGB.(img)))
julia> findmax(cat_pred_resnet)
(0.99877006f0, CartesianIndex(413, 1))
```

### LINKED ISSUE #86: Add steps to create new pre-trained models
It would be nice to document how folks can add pre-trained models to this repo

### LINKED ISSUE #143: Pre-trained models and `classify` disappeared?
About a year ago I used `Metalhead.jl` for some demonstration purposes as described here: https://juliapackages.com/p/metalhead using the `classify`-function.

Now the pre-trained models as well as the `classify`-function seem to have disappeared. What happened to this and how can I do it now?

### PR DESCRIPTION
**Pull request by darsnack**
This PR builds on the great work by @Alexander-Barth in https://github.com/FluxML/MetalheadWeights/pull/2 to add pre-trained weights for VGG and ResNet from torchvision.

We ran into issues with Github LFS limits on MetalheadWeights. Instead, we are using HuggingFace to host our models now which is really ideal for our use case.

This PR adds:
- [x] Pre-trained models for VGG (w/o BN) + ResNet
- [x] Updated quickstart guide
- [x] Updated contributor docs for how to others can add weights via HF
- [x] Refactored tests to check the accuracy of pre-trained models (by just checking against a single reference image from ImageNet)

Close #86, closes #72, and closes #143.

### CODE REVIEW

File: docs/dev-guide/contributing.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+## Adding pre-trained weights
+
+To add pre-trained weights for an existing model or new model, you can [open a PR](https://github.com/FluxML/Metalhead.jl/pulls). Below, we describe the steps you should follow to get there.
+
+1. Train your model or port the weights from another framework.
+2. Save the model using [BSON.jl](https://github.com/JuliaIO/BSON.jl) with `BSON.@save "modelname.bson" model`. It is important that your model is saved under the key `model`.
+3. Compress the saved model as a tarball using `tar -cvzf modelname.tar.gz modelname.bson`.
+4. Obtain the SHAs (see the [Pkg docs](https://pkgdocs.julialang.org/v1/artifacts/#Basic-Usage)). Edit the `Artifacts.toml` file in the Metalhead.jl repository and add entry for your model. You can leave the URL empty for now.
+5. Open a PR on Metalhead.jl. Be sure to ping a maintainer (e.g. `@darsnack`) to let us know that you are adding a pre-trained weight. We will create a model repository on HuggingFace if it does not already exist.
+6. Open a PR to the [corresponding HuggingFace repo](https://huggingface.co/FluxML). We will merge your PR.
```
Comment (CarloLucibello) [inline docs/dev-guide/contributing.md:33]: maybe this is clear to regular HuggingFace users but not to me. How do I open a PR? Should I follow these instructions https://huggingface.co/docs/transformers/model_sharing?

Comment (darsnack) [reply docs/dev-guide/contributing.md]: Good point, updated with more details and links to HF docs.

File: docs/dev-guide/contributing.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+## Fixing bugs
+
+To fix a bug in Metalhead.jl, you can [open a PR](https://github.com/FluxML/Metalhead.jl/pulls). It would be helpful to file an issue first so that we can confirm the bug.
+
+## Adding models
+
+To add a new model architecture to Metalhead.jl, you can [open a PR](https://github.com/FluxML/Metalhead.jl/pulls). Keep in mind a few guiding principles for how this package is designed:
+
+- reuse layers from Flux as much as possible (e.g. use `Parallel` before defining a `Bottleneck` struct)
+- adhere as closely as possible to a reference such as a published paper (i.e. the structure of your model should be intuitive based on the paper)
```
Comment (CarloLucibello) [inline docs/dev-guide/contributing.md:18]:
```suggestion
- adhere as closely as possible to a reference such as a published paper (i.e. the structure of your model should be intuitively based on the paper)
```

File: docs/dev-guide/contributing.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+- use generic functional builders (e.g. [`resnet`](#) is the core function that build "ResNet-like" models based on the principles in the paper)
```
Comment (CarloLucibello) [inline docs/dev-guide/contributing.md:19]:
```suggestion
- use generic functional builders (e.g. [`resnet`](#) is the core function that builds "ResNet-like" models)
```

File: docs/dev-guide/contributing.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+4. Obtain the SHAs (see the [Pkg docs](https://pkgdocs.julialang.org/v1/artifacts/#Basic-Usage)). Edit the `Artifacts.toml` file in the Metalhead.jl repository and add entry for your model. You can leave the URL empty for now.
+5. Open a PR on Metalhead.jl. Be sure to ping a maintainer (e.g. `@darsnack`) to let us know that you are adding a pre-trained weight. We will create a model repository on HuggingFace if it does not already exist.
+6. Open a PR to the [corresponding HuggingFace repo](https://huggingface.co/FluxML). Do this by going to the "Community" tab in the HuggingFace repository. PRs and discussions are shown as the same thing in the HuggingFace web app. You can use your local Git program to make clone the repo and make PRs if you wish. Check out the [guide on PRs to HuggingFace](https://huggingface.co/docs/hub/repositories-pull-requests-discussions) for more information.
+7. Copy the download URL for the model file that you added to HuggingFace. Make sure to grab the URL for a specific commit and not for the `main` branch.
+8. Update your Metalhead.jl PR by adding the URL to the Artifacts.toml.
+9. If the tests pass for your weights, we will merge your PR!
+
```
Comment (CarloLucibello) [inline docs/dev-guide/contributing.md:39]: Is there anything to be said about which kind of tests we expect to be added or how to make sure that the model is well trained?

File: test/runtests.jl [hunk 5]
```diff
@@ -10,6 +23,28 @@ function gradtest(model, input)
     return true
 end
 
+function normalize(data)
```
Comment (CarloLucibello) [inline test/runtests.jl:26]:
```suggestion
function normalize_imagenet(data)
```

File: docs/dev-guide/contributing.md (hunk 4 above)
Comment (CarloLucibello) [reply docs/dev-guide/contributing.md:39]: Ok, I see we have `acctest` testing a few images. Ideally when adding a new model also the full test accuracy should be reported, and we can keep them in a table

File: test/runtests.jl [hunk 6]
```diff
@@ -10,6 +23,28 @@ function gradtest(model, input)
...
+
+# image net labels
+const TEST_LBLS = readlines(download("https://raw.githubusercontent.com/pytorch/hub/master/imagenet_classes.txt"))
+
+function acctest(model)
+    ypred = model(TEST_X) |> vec
+    top5 = TEST_LBLS[sortperm(ypred; rev = true)]
+
+    return "acoustic guitar" ∈ top5
+end
```
Comment (CarloLucibello) [inline test/runtests.jl:46]: In future PRs we could improve this test by evaluating accuracy on a subsample of the testset, maybe 1k images or so, a bit noisy but possibly manageable by CI

Comment (darsnack) [reply test/runtests.jl:46]: Agreed or on a subset of "imagenette"? I mostly wanted to avoid trying to download the dataset on CI.

Comment (CarloLucibello) [reply test/runtests.jl:46]: yes imagenette is an option, or maybe one can download only the test dataset

Comment (darsnack) [reply test/runtests.jl:46]: Like you said, let's do that in another PR?

Comment (CarloLucibello) [reply test/runtests.jl:46]: Let's go than

**Review by CarloLucibello — APPROVED** [review]
